GitCommands/Patches/PatchManager.cs

```diff
@@ -82,6 +88,52 @@ private static string CorrectHeaderForNewFile(string header)
             return sb.ToString();
         }
 
+        private static string CorrectHeaderForRenamedFile(string header)
+        {
+            string[] headerLines = header.Split(new[] { '\n' });
+            string? oldNameFull = null;
+            string? newName = null;
+            foreach (string line in headerLines)
+            {
+                if (line.StartsWith("+++"))
+                {
+                    newName = line[6..];
+                }
+                else if (line.StartsWith("---"))
+                {
+                    oldNameFull = line[4..];
+                }
+            }
+
+            StringBuilder sb = new();
+
+            for (int i = 0; i < headerLines.Length; i++)
+            {
+                if (i != 0)
+                {
+                    sb.Append('\n');
+                }
+
+                string line = headerLines[i];
+                if (line.StartsWith("--- "))
+                {
+                    line = $"--- a/{newName}";
+                }
+                else if (line.Contains($" {oldNameFull} "))
+                {
+                    line = line.Replace($" {oldNameFull} ", $" a/{newName} ");
+                }
+                else if (line.StartsWith("rename from ") || line.StartsWith("rename to ") || line.StartsWith("similarity index "))
```

Comment:
Are any of these strings ever translated in Git?

Reply:
Since these are part of a diff format used by git and are called extended headers I think it's safe to assume that that wouldn't happen. At least documentation doesn't mention anything of that sort.
https://git-scm.com/docs/diff-format

Reply:
It is described as a format, it is not even a 'requested' format, certainly not a porcelain interface.
Adding translations in Git will break too.
It could be OK anyway, if the consequences are low (like current behavior in master).
There need to be at least comments about this.

Reply:
I'd like to push back if I may. Translation argument sounds almost like saying that translating browsers would mean translating HTTP Headers they send (custom ones, not covered by the standard).
These are headers added by the tools for the tools (at least I would assume so). They are meant to be parsable. Translating them would break older versions thus I can't imaging that ever happening. Although if you still insist, I'll add a comment.